scripts/mysql_secure_installation.sh

```diff
@@ -207,7 +207,15 @@ validate_reply () {
 	reply=y
 	return $ret
     fi
-    case $1 in
+    #
+    # Convert user input to lower case. After that we get case independent response from 
+    # the validate_reply function. if tr binary not available or corrupted on targeted host,
+    # it will fallback to previous behaviour mean case dependent behaviour.   
+    yes_or_no=$(echo $1 | tr '[:lower:]' '[:upper:]')
```

Comment:
The comment says "# Convert user input to lower case." but the `tr` command translates lower-case characters to upper-case.

Thus, if I am reading this correctly, I believe the following `$yes_or_no` would never be equal to either "yes" or "no", which will simply set `$yes_or_no` to `$1` .... which is the same as `case $1 in` ....

Reply:
You are 100% correct here, good catch I didn't dig too much into the implementation details because I'm uncertain if this is a problem that needs to be solved.

Reply:
I'm not sure I understand, as even "YES" and "NO" are not equal to "yes" or "no" and thus `$yes_or_no` would be reset to $1 .... however ... it really doesn't matter! If we wish to be fully case-insensitive, @vuvova presented a more obvious way to do so.

Reply:
> The comment says "# Convert user input to lower case." but the `tr` command translates lower-case characters to upper-case.
> 
> Thus, if I am reading this correctly, I believe the following `$yes_or_no` would never be equal to either "yes" or "no", which will simply set `$yes_or_no` to `$1` .... which is the same as `case $1 in` ....

Yes you are right , thanks for pointing out problem in my code